app/controllers/AddBusinessTradeController.scala

```diff
@@ -0,0 +1,131 @@
+import config.featureswitch.{FeatureSwitching, IncomeSources}
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.predicates.{AuthenticationPredicate, IncomeSourceDetailsPredicate, NavBarPredicate, NinoPredicate, SessionTimeoutPredicate}
+import forms.utils.SessionKeys
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import views.html.AddBusinessTrade
+import models.incomeSourceDetails.BusinessTradeForm
+import services.IncomeSourceDetailsService
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class AddBusinessTradeController @Inject()(authenticate: AuthenticationPredicate,
+                                           val authorisedFunctions: AuthorisedFunctions,
+                                           checkSessionTimeout: SessionTimeoutPredicate,
+                                           retrieveNino: NinoPredicate,
+                                           val addBusinessTradeView: AddBusinessTrade,
+                                           val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                           val retrieveBtaNavBar: NavBarPredicate,
+                                           val itvcErrorHandler: ItvcErrorHandler,
+                                           incomeSourceDetailsService: IncomeSourceDetailsService)
+                                          (implicit val appConfig: FrontendAppConfig,
+                                             implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                             implicit override val mcc: MessagesControllerComponents,
+                                             val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching{
+
+  val backURL: String = "/report-quarterly/income-and-expenses/view/income-sources/add/business-start-date-check"
+  val agentBackURL: String = "/report-quarterly/income-and-expenses/view/agents/income-sources/add/business-start-date-check"
+  def show: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(isAgent = false)
+  }
+
+  def showAgent: Action[AnyContent] =
+    Authenticated.async {
+      implicit request =>
+        implicit user =>
+          getMtdItUserWithIncomeSources(incomeSourceDetailsService, useCache = true) flatMap {
+            implicit mtdItUser =>
+              handleRequest(isAgent = true)
+          }
+    }
+
+  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    Future {
+      if (isDisabled(IncomeSources)) {
+        Redirect(controllers.routes.HomeController.show())
```

Comment:
* pls wrap Redirect in the Future.successful
and remove Future from line 71; 
+ wrap lines 75-76 in the Future

Reply:
Addressed